### DISCUSSION TITLE
FS22

### DISCUSSION THREAD

**Discussion by jiggin** [Courseplay/courseplay#7203] [Q&A] [answered]
Sorry if this has been asked elsewhere, or shouldn't be asked here (or at all!). are you guys working with Giants for fs22, or do you have to start all over again? it would be great to see it integrated like seasons has been.

Courseplay is, and always has been, a big part of my gameplay, and need to know if I should pre order 22 or not!

Regards

James

**Comment by Tensuko** [comment, marked as answer]
We are not working together with GIANTS on FS22.
However, we will try to bring Courseplay to FS22 as soon as possible.

That all depends on whats going to be new on FS22, how does the Helper look like and how much we would have to change on the code.

We don't know more about FS22 then all of you.
So patient is the key.

**Reply by pauloluz72** [reply]
bom trabalho o courseplay parabens abraco para toda a equipa

**Reply by pauloluz72** [reply]
good job with courseplay congratulations to the whole team

**Comment by MoslessNeo** [comment]
I would like to know if you (team) have any relationship with ambassadors from your respective countries;
Or they have already tried to contact the Giants team for a developmental relationship.
I see that you are excellent programmers and have a good command of the Lua language.
Like the realismusmodding team (https://www.realismusmodding.com/) they had a partnership with the Giants development team.
I see Courseplay as an immense potential!

I am very happy to see the mod being presented to ModHub, asism even more people will have access to this mod.

**Reply by Tensuko** [reply]
We are ofc in contact with GIANTS, but not the way that we are working together with them on FS22.
We all have our own job, thats why this is more a hobby project.

CP 6.04 was sent in to ModHub, lets see what we have to do so GIANTS accept it for ModHub :)

**Comment by ConstructionGamer** [comment]
Was great to see CP accepted on Modhub 🙂. Such a big part of my FS19 experience. It looks as if FS22 will have irregular shaped fields and some sort of GPS, but I imagine it won't be as comprehensive as Courseplay, so hope you are able to port it over. Hopefully it won't have to be completely rebuilt like you had to do for FS19.

**Comment by fudoido** [comment]
Seems giants has decided to implement some of the courseplay features into the game.
Hope courseplay can be developed for some extra functions that giants hasn't used, such as silo compacting.

https://www.farming-simulator.com/newsArticle.php?news_id=281

**Reply by ConstructionGamer** [reply]
Be interested to see how it works. Just had a reply from Giants about modded maps, it'll require some sort of code within the map which needs to be implemented by modders to work. For me Courseplay will still be a must for sure

**Reply by Tensuko** [reply]
We will see how we adapt to this or do our own thing again...

**Reply by fudoido** [reply]
@Tensuko can I suggest that, since you guys are working together already, courseplay for fs22 can have the help and insight from Stephan (think I spelled it right) , the author of autodrive?
Instead of working again on 2 mods, all of you work along for a improved courseplay.... Call it autoplay or coursedrive, but if you guys plan to make a fs22 version, would be great to have both the teams as one.

**Reply by Tensuko** [reply]
Time will tell...

**Reply by knoxville73** [reply]
By the way, many thanks for your great work about FS19 !

**Reply by rohne83** [reply]
Except for mode 1 and 5 I don't see any CP features which are implemented by Giants here (not speaking of algorithms behind those, that's a different story). Functionality is rather kind of a very reduced AutoDrive. So also in FS22 we would still need CP with the same functionality it has in FS19. Let's hope API and documentation allow for a quick come back of CP in FS22.
It's rather AutoDrive where the big question mark is now.

**Reply by Tensuko** [reply]
We will see what and how we do it...

**Comment by Yllelder** [comment]
This game is not worth playing without Courseplay.

Thank you very much for so many hours of fun in FS19.

You guys make 50% of this game.

**Reply by pull2play** [reply]
Agreed. Over the years I have found that I can usually spend up to 100 hours on vanilla Farming Simulator. After that point doing everything myself due to useless in-built AI becomes a tiresome slog and any fun is lost. Courseplay is an invaluable tool and the only reason I keep coming back to the franchise. I cannot imagine playing this game without it.

**Comment by fudoido** [comment]
have been posting a discussion on stephan's "autodrive" github about a fs22 version. my last comment was the same suggestion i made above, and developed my suggestion.....
Both mod creators come together, instead of creating 2 separate mods and then make them work with each other, everyone work in the same mod.
Focus on what the game AI does NOT have and try to implement it in CP/AD fs22 mod. i know fs22 AI is not the best, but i rather have a simplified CP/AD mod released soon (soonish, cant be too soon i know lol) than waiting months or years for a "complete version" with lots of features. I think we can manage fine with the current game AI, is just the features that CP/AD gave us before that we are missing, such as being able to unload from combines, silo compacting, shovel work, maybe a proper fieldwork calculation.
I also believe that, until giants editor isnt out, you wont be able to do much as i heard that the AI is working on some sort of map "mask", and to see and learn about that, you must open maps on editor, so we all have to wait for GE to be launched and then we will see proper mods coming out.

**Comment by TSO-Original** [comment]
Made another donation today. I have purchased FS22, but so far it's been an easy choice to hold off and continue my FS19 game, largely due to the enjoyment that Courseplay allows me in managing a fairly large farm. I am hopeful that Giants will release the new editor soon and that you will be able to convert Courseplay with less hassle than it was to go from FS17 to FS19.

**Reply by Tensuko** [reply]
It wont be just a convert.
But we are on it.

**Reply by Jeanmard** [reply]
Great to hear. This game is many times better once you use Courseplay. The AI is still funky in FS22.

**Comment by ConstructionGamer** [comment]
I was excited to see what Giants have done, but the AI is still useless, can't work fields which aren't perfectly square. I've uninstalled FS19 so will persevere with FS22 whilst eagerly awaiting a new Courseplay.

**Comment by jeffreytoller** [comment]
Also just made a donation. Thank you for your hard work.

**Comment by BattleHook** [comment]
Hi Guys, keep up the good work on this project. I really liked it in FS19 and can t wait to use it in FS22.

I got the impression that Courseplay was already modified to FS22 (see below) untill I read this discussion here (see above).

**Reply by Tensuko** [reply]
And why do you share this fake site ?
It is full of mod stealing and just generating clicks.
They get money for visiting this side, more interesting stuff, more money.
The only way to get rid of such sides is to ignore them!

**Reply by BattleHook** [reply]
No need to flame me. How should I know it is a fake web-site ? It seems legit. OIt actually refers to Courseplay on Github, so no stealing as far as Courseplay is concerned, maybe some pictures from your manual ?
All you had to do ( and apparently you did ) is remove the link to that website.
Anyway no intent here to hi-jack this project from my side. Just take the compliment: Courseplay is a great addition to FS22.

**Reply by Tensuko** [reply]
It wasn't even flame or something. It was to make clear, no share from any website.
All those fs22-mods or farmingsimulator22mods blah blah sites are all fake shit.
The only source you will ever find the real courseplay to download is here on github and Giants ModHub.
We know about the fake shit. But you know, just by take a look at the picture they used (old HUD, 6.01.003xx something) it is simply scam or whatever.
And we don't need to know the source or something we can't do anything about that. Just make clear where to get it from and advice ppl to not visit those websites...

**Reply by LingTheBug** [reply]
It'll be wonderful when it's updated. I agree that I was lured with those sites too, but after looking here it's clear what the truth is, so I'll star the repo and watch for updates eventually coming through!

**Reply by BattleHook** [reply]
Any idea on the timetable for CoursePlay for FS22 ??? I really look forward to this.......

**Reply by Redster999** [reply]
No, there is no timetable. And we are all waiting in suspense. Head over to issues and you can see pictures of their progress. Looks promising. Personally i think it should not be to long untill we see a decent mod from the CP team. Full package is coming later. But the generatiion of feildwork looks to be up next. We all have to sit and wait for our wish to come true ;)
